tests/utils_tests/test_timezone.py

```diff
@@ -54,6 +57,24 @@ def test_now(self):
         with override_settings(USE_TZ=False):
             self.assertTrue(timezone.is_naive(timezone.now()))
 
+    def test_localdate(self):
+        naive = datetime.datetime(2015, 1, 1, 0, 0, 1)
+        with self.assertRaisesMessage(ValueError, 'astimezone() cannot be applied to a naive datetime'):
+            timezone.localdate(naive)
+        with self.assertRaisesMessage(ValueError, 'astimezone() cannot be applied to a naive datetime'):
+            timezone.localdate(naive, timezone=EAT)
+
+        aware = datetime.datetime(2015, 1, 1, 0, 0, 1, tzinfo=ICT)
+        self.assertEqual(timezone.localdate(aware, timezone=EAT), datetime.date(2014, 12, 31))
+        with timezone.override(EAT):
+            self.assertEqual(timezone.localdate(aware), datetime.date(2014, 12, 31))
+
+        with mock.patch('django.utils.timezone.now') as mock_now:
```

Comment:
I think this works fine and is a bit simpler: `with mock.patch('django.utils.timezone.now', return_value=aware):`